server/runtime/src/main/java/org/infinispan/server/Server.java

```diff
@@ -263,15 +263,18 @@ private Server(File serverRoot, Properties properties) {
    }
 
    private void parseConfiguration(URL config) {
-      ParserRegistry parser = new ParserRegistry(Thread.currentThread().getContextClassLoader(), false, properties);
+      ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+      ParserRegistry parser = new ParserRegistry(classLoader, false, properties);
       try {
          // load the defaults first
          URL defaults = this.getClass().getClassLoader().getResource(SERVER_DEFAULTS);
          defaultsHolder = parser.parse(defaults);
 
          // base the global configuration to the default
          configurationBuilderHolder = new ConfigurationBuilderHolder();
-         configurationBuilderHolder.getGlobalConfigurationBuilder().read(defaultsHolder.getGlobalConfigurationBuilder().build());
+         configurationBuilderHolder.getGlobalConfigurationBuilder()
+               .read(defaultsHolder.getGlobalConfigurationBuilder().build())
+               .classLoader(classLoader);
```

Comment:
Wouldn't it be better to make `ParserRegistry` set the classloader automatically?